ingress-node-firewall/v1alpha1/ingressnodefirewall_webhook.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/openshift/ingress-node-firewall/pkg/failsaferules"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+type empty struct{}
+type uint32Set map[uint32]empty
+
+// log is for logging in this package.
+var (
+	ingressnodefirewalllog = logf.Log.WithName("ingressnodefirewall-resource")
+)
+
+func (r *IngressNodeFirewall) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-ingress-nodefw-ingress-nodefw-v1alpha1-ingressnodefirewall,mutating=false,failurePolicy=fail,sideEffects=None,groups=ingress-nodefw.ingress-nodefw,resources=ingressnodefirewalls,verbs=create;update,versions=v1alpha1,name=vingressnodefirewall.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &IngressNodeFirewall{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *IngressNodeFirewall) ValidateCreate() error {
```

Comment:
You should avoid having functions associated with your API package. It brings in additional dependencies and makes it difficult for people to import and re-use your API types.

You can wrap the webhook in an additional struct to avoid this pattern

Reply:
can u pls expand on this more ?

Reply:
Ill do this. Thanks Joel.

Reply:
PR: https://github.com/openshift/ingress-node-firewall/pull/81